### ISSUE TITLE
Add support for bumping the fee on RBF transactions

### ISSUE THREAD

**Issue by notmandatory** [bitcoindevkit/bdk-ffi#146]
Add options to the `TxBuilder` to support the bumping the fee on a users RBF transaction.

See https://docs.rs/bdk/latest/bdk/wallet/tx_builder/struct.TxBuilder.html for docs on what the functions do.

```udl
interface TxBuilder {
  ... 
  // if this is set then the bdk wallet.build_fee_bump(txid) function is called instead of wallet.build_tx()
  TxBuilder fee_bump(string txid); 
  // this option is only valid for fee bump transactions, should log warning if fee_bump not set
  TxBuilder allow_shrinking(string address);
}
```

This is a departure from current BDK API since otherwise I don't see a straightforward way to do it without duplicating code into a new BumpFeeTxBuilder, or something like that.

**Comment by notmandatory** [comment]
@artfuldev I'm going to start working on this one. Would love to get your review when it's ready.

**Comment by notmandatory** [comment]
I had to add a new interface for bumping fees:
```udl
interface BumpFeeTxBuilder {
  constructor(string txid, float new_fee_rate);
  BumpFeeTxBuilder allow_shrinking(string address);
  BumpFeeTxBuilder enable_rbf();
  BumpFeeTxBuilder enable_rbf_with_sequence(u32 nsequence);
  [Throws=BdkError]
  PartiallySignedBitcoinTransaction build([ByRef] Wallet wallet);
};
```

**Comment by notmandatory** [comment]
fixed by #150
